src/frontend/http.rs

```diff
@@ -304,6 +245,60 @@ pub async fn upload(
     warp::reply::with_status(Ok("done"), StatusCode::OK).into_response()
 }
 
+async fn load_part(p: Part) -> Result<Vec<u8>, warp::Rejection> {
+    p.stream()
+        .try_fold(Vec::new(), |mut vec, data| {
+            vec.put(data);
+            async move { Ok(vec) }
+        })
+        .await
+        .map_err(|e| {
+            eprintln!("Error reading part's data: {}", e);
+            warp::reject::reject()
+        })
+}
+
+fn load_csv_bytes(
+    source: Vec<u8>,
+    schema: Option<Schema>,
+    has_header: bool,
+) -> Result<(Schema, Vec<RecordBatch>), ArrowError> {
+    // If the schema part wasn't specified we'll need to infer it
+    let builder = match schema {
+        Some(schema) => ReaderBuilder::new()
+            .with_schema(Arc::new(schema))
+            .has_header(has_header),
+        None => ReaderBuilder::new()
+            .infer_schema(None)
+            .has_header(has_header),
+    };
+
+    let mut cursor = Cursor::new(source);
+    let csv_reader = builder.build(&mut cursor).unwrap();
+    let schema = csv_reader.schema().as_ref().clone();
+    let partition: Vec<RecordBatch> = csv_reader
+        .into_iter()
+        .collect::<Result<Vec<RecordBatch>, ArrowError>>()
+        .unwrap();
+
+    Ok((schema, partition))
+}
+
+fn load_parquet_bytes(source: Vec<u8>) -> Result<(Schema, Vec<RecordBatch>), ArrowError> {
+    let mut parquet_reader =
+        ParquetFileArrowReader::try_new(Bytes::from(source)).unwrap();
+
+    let schema = parquet_reader.get_schema().unwrap();
+
+    let partition: Vec<RecordBatch> = parquet_reader
+        .get_record_reader(1024)
+        .unwrap()
+        .collect::<Result<Vec<RecordBatch>, ArrowError>>()
+        .unwrap();
```

Comment:
Can use `?` now since you're returning an ArrowError and it'll propagate it, might be possible to even avoid the turbofish since you have the type specified in the function signature:

```suggestion
    let mut parquet_reader =
        ParquetFileArrowReader::try_new(Bytes::from(source))?;

    let schema = parquet_reader.get_schema()?;

    let partition: Vec<RecordBatch> = parquet_reader
        .get_record_reader(1024)?
        .collect?;
```